src/Ajaxray/PHPWatermark/Watermark.php

```diff
@@ -212,4 +225,58 @@ private function ensureWritable(string $dirPath): void
             throw new \InvalidArgumentException("The specified destination $dirPath is not writable!");
         }
     }
+
+     /**
+     *  Prepare color for watermark command line
+     */
+    private function _prepareColorForWaterMark($color)
+    {
+        $hexColor = $this->_colorToRGB($color);
+
+        $opecity = $this->options['opacity'];
+
+        $rgba = "rgba\\(" . $hexColor[0] . "," . $hexColor[1] . "," . $hexColor[2] . "," . $opecity . "\\)";
+        return $rgba;
+    }
+    
+    private function _colorToRGB($hex)
+    {
+        $hex = strtolower($hex);
+
+        if (strpos($hex, 'rgba') !== false) {
+            preg_match('/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/i', $hex, $rgb);
+
+            if ($rgb) {
+                if (!empty($rgb[4]) && \XF::options()->up_wm_opecity === 0) {
```

Comment:
XF is [not found](https://www.php.net/manual-lookup.php?pattern=XF&amp;lang=en). Can you please use an alternative that does not require additional installation?

Reply:
I'm sorry for that mistake maybe i forget to change that line because i use it in my project the line number 250 should be like this:

`if (!empty($rgb[4])) {`